third-party/rsutils/src/ip-address.cpp

```diff
@@ -0,0 +1,58 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2024 Intel Corporation. All Rights Reserved.
+
+#include <rsutils/string/ip-address.h>
+
+#include <rsutils/json.h>
+using rsutils::json;
+
+
+namespace rsutils {
+namespace string {
+
+
+ip_address::ip_address( std::string str )
+{
+    char const * pch = str.c_str();
+    char const * const end = pch + str.length();
+    while( *pch >= '0' && *pch <= '9' )
+        ++pch;
+    if( *pch++ != '.' )
+        return;
```

Comment:
I think we should also check for validity and throw if not containing 4 numbers in the [0 255] range.

Reply:
Right, will do

Reply:
Done

Reply:
It does not throw -- if the IP is invalid, is_valid() will return false